code/game/objects/buckling.dm

```diff
@@ -0,0 +1,98 @@
+
+
+/obj
+	var/can_buckle = 0
+	var/buckle_lying = -1 //bed-like behaviour, forces mob.lying = buckle_lying if != -1
+	var/buckle_requires_restraints = 0 //require people to be handcuffed before being able to buckle. eg: pipes
+	var/mob/living/buckled_mob = null
+
+
+//Interaction
+/obj/attack_hand(mob/living/user)
+	. = ..()
+	if(can_buckle && buckled_mob)
+		user_unbuckle_mob(user)
+
+/obj/MouseDrop_T(mob/living/M, mob/living/user)
+	. = ..()
+	if(can_buckle && istype(M))
+		user_buckle_mob(M, user)
+
+
+//Cleanup
+/obj/Destroy()
+	. = ..()
+	unbuckle_mob()
+
+/obj/Del()
+	. = ..()
+	unbuckle_mob()
+
+
+//procs that handle the actual buckling and unbuckling
+/obj/proc/buckle_mob(mob/living/M)
+	if(!can_buckle || !istype(M) || (M.loc != loc) || M.buckled || (buckle_requires_restraints && !M.restrained()))
+		return 0
+
+	M.buckled = src
+	M.dir = dir
+	buckled_mob = M
+	M.update_canmove()
+	post_buckle_mob(M)
+	return 1
+
+/obj/proc/unbuckle_mob()
+	if(buckled_mob && buckled_mob.buckled == src)
+		. = buckled_mob
+		buckled_mob.buckled = null
+		buckled_mob.anchored = initial(buckled_mob.anchored)
+		buckled_mob.update_canmove()
+		buckled_mob = null
+
+		post_buckle_mob(.)
+
+
+//Handle any extras after buckling/unbuckling
+//Called on buckle_mob() and unbuckle_mob()
+/obj/proc/post_buckle_mob(mob/living/M)
+	return
+
+
+//Wrapper procs that handle sanity and user feedback
+/obj/proc/user_buckle_mob(mob/living/M, mob/user)
+	if(!user.Adjacent(M) || user.restrained() || user.lying || user.stat)
```

Comment:
Not being able to buckle someone else because you are buckled is a bit silly, if that buckle required restraints though you'd be restrained(), and that's the only situation that would make sense.